quipucordsctl/commands/reset_admin_password.py

```diff
@@ -0,0 +1,135 @@
+"""Reset the server's login password."""
+
+import argparse
+import difflib
+import getpass
+import logging
+import subprocess
+from gettext import gettext as _
+
+logger = logging.getLogger(__name__)
+PODMAN_SECRET_NAME = "quipucords-server-password"  # noqa: S105
+PASSWORD_MIN_LENGTH = 10
+PASSWORD_BLOCKLIST = ["dscpassw0rd", "qpcpassw0rd"]
+DEFAULT_USERNAME = "admin"
+PASSWORD_USERNAME_MAX_SIMILARITY = 0.7
+
+
+def get_help() -> str:
+    """Get the help/docstring for this command."""
+    return _("Reset the server's login password.")
+
+
+def server_password_is_set() -> bool:
+    """Check if the server password is already set."""
+    # TODO `podman secret exists quipucords-server-password`
+    return False
+
+
+def check_password(new_password, confirm_password):
+    """Check if the new password inputs are sufficient to set."""
+    if new_password != confirm_password:
+        logger.error(_("Your password inputs do not match."))
+        return False
+    if len(new_password) < PASSWORD_MIN_LENGTH:
+        # mimic MinimumLengthValidator on the server
+        logger.error(_("Your password must be at least 10 characters long."))
+        return False
+    if new_password.isdigit():
+        # mimic NumericPasswordValidator on the server
+        logger.error(_("Your password cannot be entirely numeric."))
+        return False
+    if new_password in PASSWORD_BLOCKLIST:
+        # mimic CommonPasswordValidator on the server
+        logger.error(_("Your password cannot be used because it is blocked."))
+        return False
+    if (
+        difflib.SequenceMatcher(a=new_password, b=DEFAULT_USERNAME).quick_ratio()
+        >= PASSWORD_USERNAME_MAX_SIMILARITY
+    ):
+        # mimic UserAttributeSimilarityValidator on the server
+        # TODO Refactor this when we allow the user to set a custom login username.
+        logger.error(_("Your password is too similar to your login username."))
+        return False
+
+    return True
+
+
+def prompt_password() -> str | None:
+    """Prompt the user to enter a new password."""
+    try:
+        new_password = getpass.getpass(_("Enter new server login password: "))
+        confirm_password = getpass.getpass(_("Confirm new server login password: "))
+    except KeyboardInterrupt:  # can occur via control-c input
+        # TODO move this higher in the stack
+        logger.error(_("Exiting due to keyboard interrupt."))
+        return None
+    except EOFError:  # can occur via control-d input
+        logger.error(_("Input closed unexpectedly when prompting for password."))
+        return None
```

Comment:
It is already on TODO, so I will ignore it for now but, yeah, this should be handled at the entrypoint level or as close as possible. 

As a suggestion, we could have internal Exceptions that could carry custom messages up in the stack
(or not, logger.error is probably enough for that)